WalletWasabi/BranchNBound/BranchAndBound.cs

```diff
@@ -0,0 +1,169 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.BranchNBound
+{
+	public class BranchAndBound
+	{
+		private readonly Random _random = new();
+
+		public BranchAndBound(List<Money> availableCoins)
+		{
+			Count = availableCoins.Count;
+			SortedUTXOs = availableCoins.OrderByDescending(x => x.Satoshi).Select(c => c.Satoshi).ToArray();
+		}
+
+		private enum NextAction
+		{
+			AandB,
+			BandA,
```

Comment:
This should be documented a bit more:

```suggestion
            /// <summary>First try to include a coin and then try not to include the coin in the selection.</summary>
			AandB,
            /// <summary>First try NOT to include a coin and then try to include the coin in the selection.</summary>
			BandA,
```

or renamed so that it is self-explanatory (nothing comes to my mind atm).